#109 feat: 쿠폰 관리 View API 구현 외 (rel. #119, #135)

matrixpower1004 opened the pull request:
### 💡Motivation
- 쿠폰 관리 화면 View API 구현
- 쿠폰 추가 구매 API 구현
- 쿠폰 수정 및 삭제 API 구현

### 📌Changes
- 쿠폰 관리 화면 응답 API 구현
- 쿠폰 관리 화면 응답 Test 작성
- 쿠폰 관리 화면 응답 REST Docs 작성
- 쿠폰 추가 구매 API 구현
- 쿠폰 추가 구매 Test 작성
- 쿠폰 추가 구매 REST Docs 작성
- 쿠폰 수정 및 삭제 API 구현
- 쿠폰 수정 및 삭제 Test 작성
- 쿠폰 수정 및 삭제 REST Docs 작성

### 🫱🏻‍🫲🏻To Reviewers
- 코드 리뷰 부탁 드립니다.

closes #109 
closes #119 
closes #135

matrixpower1004 committed:
#109 feat: 쿠폰 관리 화면(View data) 기능 개발

- 쿠폰 관리를 위한 View 응답용 APi 구현 완료

matrixpower1004 committed:
#109 test: 쿠폰 관리 화면 View 응답 테스트

matrixpower1004 committed:
#109 test: 쿠폰 관리 REST Docs test

matrixpower1004 committed:
#109 docs: REST Docs 문서 작성

matrixpower1004 committed:
#109 refactor: 일부 변수명 rename 및 import 정리

matrixpower1004 committed:
#109 refactor: buil 실패 나오는 테스트 수정

JeongUijeong commented on src/main/java/com/backoffice/upjuyanolja/domain/coupon/controller/CouponBackofficeController.java:
```diff
@@ -71,9 +72,26 @@ public ResponseEntity<SuccessResponse<Object>> createCouponsByRooms(
+        return ApiResponse.success(
+            HttpStatus.OK,
+            SuccessResponse.<CouponManageResponse>builder()
+                .message("쿠폰 조회에 성공하였습니다.")
```
responseRoomsView의 응답 message와 같은데 의도하신 건가요??

matrixpower1004 replied:
네. 객실별로 쿠폰 현황만 내려주는 것이라 조회라고 생각해서 메시지를 통일했습니다.

JeongUijeong replied:
넵

JeongUijeong commented on src/main/java/com/backoffice/upjuyanolja/domain/coupon/dto/response/backoffice/CouponManageQueryDto.java:
```diff
@@ -0,0 +1,26 @@
+import lombok.Builder;
+
+@Builder
+public record CouponManageQueryDto(
+
+    Long accommodationId,
```
공백 제거 부탁 드려요~~
```suggestion
public record CouponManageQueryDto(
    Long accommodationId,
```

matrixpower1004 replied:
네. 수정해서 다시 올렸습니다.

JeongUijeong commented on src/main/java/com/backoffice/upjuyanolja/domain/coupon/dto/response/backoffice/CouponInfo.java:
```diff
@@ -0,0 +1,39 @@
+import lombok.Builder;
+
+@Builder
+public record CouponInfo(
+
+    Long couponId,
```
공백 제거 부탁 드려요!
```suggestion
public record CouponInfo(
    Long couponId,
```

matrixpower1004 replied:
수정해서 다시 올렸습니다.

JeongUijeong commented on src/main/java/com/backoffice/upjuyanolja/domain/coupon/dto/response/backoffice/CouponManageRooms.java:
```diff
@@ -0,0 +1,32 @@
+    Long roomId,
+    String roomName,
+    int roomPrice,
+    List<CouponInfo> coupons
+
+) {
```
공백 제거 부탁 드려용
```suggestion
    List<CouponInfo> coupons
) {
```

matrixpower1004 replied:
수정해서 다시 올렸습니다.

JeongUijeong reviewed (changes requested):
수고하셨습니다! 😊👍🏻

chaewon12 reviewed (approved):
수고하셨습니다 👍

matrixpower1004 committed:
Merge branch 'develop' into feature/coupon-manage

matrixpower1004 committed:
#109 refactor: 코드 공백 제거

matrixpower1004 committed:
Merge branch 'feature/coupon-manage' of github.com:Upjuyanolja/Upjuyanolja_BE into feature/coupon-manage

matrixpower1004 committed:
#119 feat: 쿠푼 추가 구매 API 개발

matrixpower1004 committed:
#119 test: 쿠폰 추가 구매 API 테스트

matrixpower1004 committed:
#119 test: 쿠폰 추가 구매  REST Docs 테스트

matrixpower1004 committed:
#119 docs: 쿠폰 추가 구매 REST Docs

matrixpower1004 committed:
#135 feat: 쿠폰 수정 및 삭제 API 구현

matrixpower1004 committed:
#135 test: 쿠폰 수정 및 삭제 테스트

matrixpower1004 committed:
#135 test: 쿠폰 수정 및 삭제 REST Dcos 테스트

matrixpower1004 committed:
#135 docs: 쿠폰 수정 및 삭제 REST Docs

matrixpower1004 committed:
#135 refactor: 불필요한 코드삭제 및 일부 수정

- DiscountType 코드가 예전의 Enum에서 데이터를 읽어오는 코드로 rollback 되어 있어서 propertiy에서 값을 읽어오도록 수정
- 불필요한 초기 테스트 데이터를 삽입하는 로직 제거
- 빈 import 정리

JeongUijeong reviewed (approved):
수고하셨습니다!! 👍🏻
